js/private/js_image_layer.bzl

```diff
@@ -500,18 +531,45 @@ js_image_layer_lib = struct(
             default = "0:0",
         ),
         "compression": attr.string(
-            doc = "Compression algorithm. Can be one of `gzip`, `none`.",
-            values = ["gzip", "none"],
+            doc = "Compression algorithm. See https://github.com/bazel-contrib/bazel-lib/blob/bdc6ade0ba1ebe88d822bcdf4d4aaa2ce7e2cd37/lib/private/tar.bzl#L29-L39",
+            values = tar_lib.common.accepted_compression_types + ["none"],
             default = "gzip",
         ),
         "platform": attr.label(
             doc = "Platform to transition.",
         ),
         "generate_empty_layers": attr.bool(
-            doc = """Generate layers even if they are empty.
+            # TODO(3.0): remove this attribute.
+            doc = """DEPRECATED. An empty layer is always generated if the layer group have no matching files.""",
+            default = False,
+        ),
+        "layer_groups": attr.string_dict(
+            doc = """Layer groups to create.
+These are utilized to categorize files into distinct layers, determined by their respective paths. 
+The expected format for each entry is "<key>": "<value>", where <key> MUST be a valid Bazel and 
+JavaScript identifier (alphanumeric characters),  and <value> MAY be either an empty string (signifying a universal match)
+or a valid regular expression.
+ 
+For better performance, it is recommended to split the large parts of a js_binary to have a separate layer.
+
+The matching order for layer groups is as follows:
 
-Helpful when using js_image_layer with rules_docker.
-See https://github.com/aspect-build/rules_js/pull/1714 for more info""",
+1. `layer_groups` are checked in order first
+2. If no match is found for `layer_groups`, the `default layer groups` are checked.
+3. Any remaining files are placed into the app layer.
+
+The default layer groups are as follows and always created.
+
+```
+{
+    "node": ".*\\/js\\/private\\/node-patches.*|.*\\/bin\\/nodejs\\/.*",
```

Comment:
This `*`s here make things bold in the rendered .md. Do `*`s need to be escaped even when they are in the triple-backtick?

Reply:
eh startdoc is stupid.